File: python/design-patterns/interpreter/naive/naive_eval.py
class EquationEvaluator:
    def evaluate(self, expression: str, variables: dict) -> float:
        # ❌ dangerous python eval() calls
        parsed = expression
        for key, val in variables.items():
            parsed = parsed.replace(key, str(val))
        return eval(parsed) # ⚠️ Security vulnerability!

File: typescript/design-patterns/interpreter/naive/naive-eval.ts
// Naive approach: using eval() or fragile string parsers
class EquationEvaluator {
  evaluate(expression: string, variables: Record<string, number>): number {
    // ❌ Using eval() is a massive security hazard and lacks structural compiler safety
    let parsed = expression;
    for (const [key, value] of Object.entries(variables)) {
      parsed = parsed.replaceAll(key, value.toString());
    }
    return eval(parsed); // ⚠️ Security risk: injection attacks!
  }
}